h/subscribers.py

```diff
@@ -30,6 +31,10 @@ def publish_annotation_event(event):
     annotation_dict = presenters.AnnotationJSONPresenter(
         event.request, event.annotation).asdict()
 
+    before_save_event = AnnotationBeforeSaveEvent(event.request,
+                                                  annotation_dict)
+    event.request.registry.notify(before_save_event)
```

Comment:
I'm not super happy with the naming here since we are not _saving_ the annotation, but still need to allow `h.nipsa` to hook in and add the `nipsa` field when necessary.